### PR TITLE
feat(vpn): add data_source_vpn_connections

### PR DESCRIPTION
**Pull request by wangyuancheng01**
**What this PR does / why we need it**:add data_source_vpn_connections

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed
root@ecs-wangyuancheng:/home/huawei/go/src/github.com/huaweicloud/terraform-provider-huaweicloud (vpn)$ make testacc TEST="./huaweicloud/services/acceptance/vpn" TESTARGS="-run TestAccVPNConnectionsDataSource_Basic"
```
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpn -v -run TestAccVPNConnectionsDataSource_Basic -timeout 360m -parallel 4
=== RUN   TestAccVPNConnectionsDataSource_Basic
=== PAUSE TestAccVPNConnectionsDataSource_Basic
=== CONT  TestAccVPNConnectionsDataSource_Basic
--- PASS: TestAccVPNConnectionsDataSource_Basic (336.43s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpn       336.523s
```

### CODE REVIEW

File: docs/data-sources/vpn_connections.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+* `name` - (Optional, String) Specifies the name of the VPN connection.
+
+* `gateway_id` - (Optional, String) Specifies the gateway ID of the VPN connection.
+
+* `gateway_ip` - (Optional, Int) Specifies the gateway IP of the VPN connection.
+
+* `status` - (Optional, String) Specifies the status of the VPN connection, the value can be **ERROR** or **ACTIVE** or **DOWN** or **PENDING_CREATE** or **PENDING_UPDATE** or **PENDING_.DELETE** or **FREEZED** or **UNKNOWN**.
+
+* `vpn_type` - (Optional, String) Specifies the VPN type of the VPN connection, the value can be **POLICY** or **STATIC** or **BGP** or **POLICY-TEMPLATE**.
```
Comment (Jason-Zhang9309) [inline docs/data-sources/vpn_connections.md]: this values should be lower case

File: docs/data-sources/vpn_connections.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+<a name="connections"></a>
+The `connections` block supports:
+
+* `id` - Indicates the ID of the connections.
+
+* `name` - Indicates the name of the connections.
+
+* `status` - Indicates the status of the connections, the value can be **ERROR** or **ACTIVE** or **DOWN** or **PENDING_CREATE** or **PENDING_UPDATE** or **PENDING_.DELETE** or **FREEZED** or **UNKNOWN**.
+
+* `vpn_type` - Indicates the VPN type of the connections, the value can be **POLICY** or **STATIC** or **BGP** or **POLICY-TEMPLATE**.
```
Comment (Jason-Zhang9309) [inline docs/data-sources/vpn_connections.md]: this values should be lower case

File: docs/data-sources/vpn_connections.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+* `id` - The data resource ID.
+
+* `connections` - All resource connection that match the filter parameters.
+  The [connections](#connections) structure is documented below.
+
+<a name="connections"></a>
+The `connections` block supports:
+
+* `id` - Indicates the ID of the connections.
```
Comment (Jason-Zhang9309) [inline docs/data-sources/vpn_connections.md]:
```suggestion
* `id` - Indicates the ID of the connection.
```

File: docs/data-sources/vpn_connections.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+<a name="dpd"></a>
+The `ikedpdpolicy` block supports:
+
+* `timeout` - Indicates the timeout of the dpd certificate, the value ranges from **2** to **60** seconds.
+
+* `interval` - Indicates the interval of the dpd certificate, the value ranges from **10** to **3600** seconds.
+
+* `msg` - Indicates the msg of the dpd certificate, the value can be **seq-hash-notify** or **seq-notify-hash**.
+
+
```
Comment (Jason-Zhang9309) [inline docs/data-sources/vpn_connections.md]: too many blank lines

File: huaweicloud/services/vpn/data_source_huaweicloud_vpn_connections.go [hunk 5]
```diff
@@ -0,0 +1,411 @@
...
+		"ikepolicy":             flattenGetConnectionIkePolicy(cnt),
+		"ipsecpolicy":           flattenGetConnectionIpsecPolicy(cnt),
+	}
+	return resourceConnections
+}
+
+func flattenGetConnectionIkePolicy(cnt connections.Connections) []interface{} {
+	ikePly := cnt.Ikepolicy
+	var rst []interface{}
+	rst = []interface{}{
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/vpn/data_source_huaweicloud_vpn_connections.go]:
```suggestion
	rst := []interface{}{
```

File: huaweicloud/services/vpn/data_source_huaweicloud_vpn_connections.go [hunk 6]
```diff
@@ -0,0 +1,411 @@
...
+			"dpd":                      flattenGetConnectionDPD(cnt),
+		},
+	}
+	return rst
+}
+
+func flattenGetConnectionDPD(cnt connections.Connections) []interface{} {
+	dpdConncetion := cnt.Ikepolicy.Dpd
+	var rst []interface{}
+	rst = []interface{}{
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/vpn/data_source_huaweicloud_vpn_connections.go]:
```suggestion
	rst := []interface{}{
```

File: huaweicloud/services/vpn/data_source_huaweicloud_vpn_connections.go [hunk 7]
```diff
@@ -0,0 +1,411 @@
...
+			"msg":      dpdConncetion.Msg,
+		},
+	}
+	return rst
+}
+
+func flattenGetConnectionIpsecPolicy(cnt connections.Connections) []interface{} {
+	ipsecPly := cnt.Ipsecpolicy
+	var rst []interface{}
+	rst = []interface{}{
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/vpn/data_source_huaweicloud_vpn_connections.go]:
```suggestion
	rst := []interface{}{
```

File: huaweicloud/services/vpn/data_source_huaweicloud_vpn_connections.go [hunk 8]
```diff
@@ -0,0 +1,411 @@
...
+			"transform_protocol":       ipsecPly.TransformProtocol,
+			"encapsulation_mode":       ipsecPly.EncapsulationMode,
+		},
+	}
+	return rst
+}
+
+func flattenGetConnectionPolicyRule(cnt connections.Connections) []interface{} {
+	plyRule := cnt.PolicyRules
+	curJson := utils.PathSearch("vpn_connection.policy_rules", plyRule, make([]interface{}, 0))
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/vpn/data_source_huaweicloud_vpn_connections.go]: PathSearch should be used here

File: docs/data-sources/vpn_connections.md (hunk 1 above)
Comment (wangyuancheng01) [reply docs/data-sources/vpn_connections.md]: down

File: docs/data-sources/vpn_connections.md (hunk 2 above)
Comment (wangyuancheng01) [reply docs/data-sources/vpn_connections.md]: down

File: docs/data-sources/vpn_connections.md (hunk 3 above)
Comment (wangyuancheng01) [reply docs/data-sources/vpn_connections.md]: down

File: docs/data-sources/vpn_connections.md (hunk 4 above)
Comment (wangyuancheng01) [reply docs/data-sources/vpn_connections.md]: down

File: huaweicloud/services/vpn/data_source_huaweicloud_vpn_connections.go (hunk 5 above)
Comment (wangyuancheng01) [reply huaweicloud/services/vpn/data_source_huaweicloud_vpn_connections.go]: down

File: huaweicloud/services/vpn/data_source_huaweicloud_vpn_connections.go (hunk 6 above)
Comment (wangyuancheng01) [reply huaweicloud/services/vpn/data_source_huaweicloud_vpn_connections.go]: down

File: huaweicloud/services/vpn/data_source_huaweicloud_vpn_connections.go (hunk 7 above)
Comment (wangyuancheng01) [reply huaweicloud/services/vpn/data_source_huaweicloud_vpn_connections.go]: down

File: huaweicloud/services/vpn/data_source_huaweicloud_vpn_connections.go (hunk 8 above)
Comment (wangyuancheng01) [reply huaweicloud/services/vpn/data_source_huaweicloud_vpn_connections.go]: down

File: huaweicloud/services/vpn/data_source_huaweicloud_vpn_connections.go [hunk 9]
```diff
@@ -0,0 +1,406 @@
...
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func ConnectionIkePolicySchema() *schema.Resource {
```
Comment (ShiChangkuo) [inline huaweicloud/services/vpn/data_source_huaweicloud_vpn_connections.go]: use lowercase will be fine, please modify others.

```suggestion
func connectionIkePolicySchema() *schema.Resource {
```

Comment (wangyuancheng01) [reply huaweicloud/services/vpn/data_source_huaweicloud_vpn_connections.go]: down

File: docs/data-sources/vpn_connections.md [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+* `gateway_ip` - Indicates the gateway IP of the connection.
+
+* `customer_gateway_id` - Indicates the customer gateway ID of the connection.
+
+* `peer_subnets` - Indicates the peer subnets of the connection.
+
+* `tunnel_local_address` - Indicates the tunnel local address of the connection.
+
+* `tunnel_peer_address` - Indicates the tunnel peer address of the connections.
```
Comment (ShiChangkuo) [inline docs/data-sources/vpn_connections.md]:
```suggestion
* `tunnel_peer_address` - Indicates the tunnel peer address of the connection.

```

File: docs/data-sources/vpn_connections.md [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+* `customer_gateway_id` - Indicates the customer gateway ID of the connection.
+
+* `peer_subnets` - Indicates the peer subnets of the connection.
+
+* `tunnel_local_address` - Indicates the tunnel local address of the connection.
+
+* `tunnel_peer_address` - Indicates the tunnel peer address of the connection.
+
+
+* `enable_nqa` - Indicates the enable nqa of the connection.
```
Comment (ShiChangkuo) [inline docs/data-sources/vpn_connections.md]:
```suggestion
* `tunnel_peer_address` - Indicates the tunnel peer address of the connection.

* `enable_nqa` - Indicates the enable nqa of the connection.
```

**Pull request merged**